[FEATURE] Ajout de l'étape de conflit lors de la récupération de compte à la sortie du SCO (PIX-2730).

Anne-Gaelle-S opened the pull request:
## :unicorn: Problème
Dans le cadre de la sortie SCO, il faut permettre à l'utilisateur de récupérer son compte Pix.
Cependant dans certains cas on ne peut pas leur permettre de le faire seuls et il faut qu'ils puissent contacter le support pour récupérer leur compte.
Par exemple, le cas où on ne permet pas à l'utilisateur de récupérer son compte, c'est lorsqu'on trouve un autre étudiant relié à ce même compte. Ca arrive notamment quand un frère et une soeur utilisent le même compte pour faire leur exercices sur Pix.

## :robot: Solution
Afficher une page d'erreur quand on trouve plusieurs schooling-registration associé au compte que l'utilisateur essaie de récupérer.

## :rainbow: Remarques
- On a pas réussit à passer à une nouvelle route `conflict` un paramètre (firstName en l'occurrence)
En effet, un transitionTo (ou replaceWith) requière un model OU un id en paramètre et rien d'autre.

- On a donc choisit de remplacer la route `conflict` par un component (car il est plus simple de lui filer un paramètre firstName)

- Il n'y aura pas d'ajout de seeds pour tester cette PR car la feature n'est pas prévue d'être souvent modifiée.

## :100: Pour tester

Sur la RA : 
- aller sur `/recuperer-mon-compte` (sur pix app)
- rentrer les infos d'utilisateur qui ont des INE différents :
Ine/Ina : 123456789YY
Prénom : Lyanna
Nom : Mormont
Date de naissance : 07/01/2002

- constater qu'on arrive sur la page d'erreur avec les textes suivants (le CSS sera fait ultérieurement) : 
<img width="293" alt="image" src="https://user-images.githubusercontent.com/38167520/123825110-ea5a2080-d8fe-11eb-8166-e072751c631f.png">

En local : 
Modifier les seeds pour qu'un même user ait 2 schooling registration différents et rejouer le scénario.

chrc committed:
[APP] Add validation and use controller method in the component

chrc committed:
[APP] Add translations in french

pix-service commented:
I'm deploying this PR to these urls:

- App (.fr): https://app-pr3144.review.pix.fr
- App (.org): https://app-pr3144.review.pix.org
- Orga: https://orga-pr3144.review.pix.fr
- Certif: https://certif-pr3144.review.pix.fr
- Admin: https://admin-pr3144.review.pix.fr
- API: https://api-pr3144.review.pix.fr/api/

 Please check it out!

yahyayouness reviewed (commented):
Le feature-toggle IS_SCO_ACCOUNT_RECOVERY_ENABLED a été activé

yahyayouness reviewed (commented):
{"errors":[{"status":"400","title":"Bad Request","detail":"\"data.attributes.birthdate\" must be in YYYY-MM-DD format"}]}

Avec une date valide coté front, la validation JOI pose problème

La date en question 1900-1-1 n'est pas au bon format

yahyayouness commented on mon-pix/app/routes/recover-account-after-leaving-sco/conflict.js:
```diff
@@ -0,0 +1,9 @@
+export default class RecoverAccountAfterLeavingScoConflictRoute extends Route {
+
+  async beforeModel(transition) {
+    console.log(transition);
```
remove console.log

chrc committed:
[APP] Add POST ../recover-account in mirage

Add POST /schooling-registration-dependent-users/recover-account in mirage configuration

chrc committed:
[APP] Add recover account conflict error component

chrc committed:
[APP] Add conflict-error component

chrc committed:
[APP] Handle api call into the route controller

AndreiaPena committed:
[APP] Add validation for dayOfBirth, monthOfBirth, yearOfBirth and birthdate format

AndreiaPena commented:
> {"errors":[{"status":"400","title":"Bad Request","detail":""data.attributes.birthdate" must be in YYYY-MM-DD format"}]}
> 
> Avec une date valide coté front, la validation JOI pose problème
> 
> La date en question 1900-1-1 n'est pas au bon format

done !

yahyayouness commented on mon-pix/app/controllers/recover-account-after-leaving-sco.js:
```diff
@@ -0,0 +1,31 @@
+    if (status === '409') {
+      this.showRecoverAccountStudentInformationForm = false;
+      this.showRecoverAccountConflictError = true;
+    }
```
Si c'est une autre error que le 409, on perd l'exception ici.

Proposition soit de les gérer , soit de refaire un throw en attendant de les gérer tous.
Exemple: Api indisponible , l'écran reste figé sans feedback à l'utilisateur

AndreiaPena replied:
On le fera sur la prochaine PR :)

yahyayouness commented on mon-pix/app/components/recover-account-student-information-form.js:
```diff
@@ -37,31 +38,64 @@ class FirstNameValidation {
+
+  get _formatBirthdate() {
+    const year = parseInt(this.yearOfBirth);
+    const month = parseInt(this.monthOfBirth) - 1;
```
Ca mérite une explication ce -1, pas assez explicite
